### ISSUE TITLE
FaceID expo-local-authentication doesn't work

### ISSUE THREAD

**Issue by TomasSestak** [expo/expo#21694]
### Minimal reproducible example

https://github.com/TomasSestak/face-id-issue

### Summary

When trying to use FaceID - it always returns "FaceID is available but has not been configured. To enable FaceID, provide `NSFaceIDUsageDescription`."

I tried both using
`    "plugins": [
      [
        "expo-local-authentication",
        {
          "faceIDPermission": "Allow $(PRODUCT_NAME) to use Face ID."
        }
      ]
    ],`
and

`ios.infoPlist.NSFaceIDUsageDescription` but with the same result

### Environment

 expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.2.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 9.1.2 - ~/.yarn/bin/npm
      Watchman: 2023.02.27.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: Electric Eel 2022.1.1 Patch 2 Electric Eel 2022.1.1 Patch 2
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~48.0.6 => 48.0.6 
      react: 18.2.0 => 18.2.0 
      react-native: 0.71.3 => 0.71.3 
    npmGlobalPackages:
      expo-cli: 6.0.8
    Expo Workflow: managed

**Comment by TomasSestak** [comment]
> Having the same issue on a bare RN project, reverting to 13.0.2 temporarily fixed the issue

can confirm - and thank you

**Comment by cosmascj** [comment]
currently facing the same issues on Bare RN project version 13.2.1

**Comment by a-leblond** [comment]
Hi !

The problem has been fixed but has not been published yet from what I've seen.
Here is the fix : https://github.com/expo/expo/commit/5a23fbd5c411cbc28adb8582617d1c41873a05e3

Any plan to publish it ?

**Comment by gasolin** [comment]
@brentvatne its a blocker for all people use iOS FACE ID auth, and the fix seems ready as @a-leblond mentioned, could you help publish it?

**Comment by enchorb** [comment]
Any ETA on the publish?

**Comment by 1stNaNo** [comment]
Any update on this problem > ?

**Comment by bernborgess** [comment]
I am also waiting for updates, since my app is on hold because of the glitch

**Comment by proctologg** [comment]
fix it, please 🥲

**Comment by brentvatne** [comment]
I have published the fix from https://github.com/expo/expo/commit/5a23fbd5c411cbc28adb8582617d1c41873a05e3 to the latest version used by sdk-48: `expo-local-authentication@~13.3.0`

apologies for the delay on this

**Comment by gregdburns** [comment]
Confirmed that the issue is resolved, thank you @brentvatne !

**Comment by evgenyshenets91** [comment]
doesn't work for me on latest version.

**Comment by cosmascj** [comment]
> doesn't work locally for me on the latest version.

version 13.3 is stable. you can consider using it or state the specific issue you are currently facing

**Comment by evgenyshenets91** [comment]
Thanks for your response. What I found out: 
- latest version works after running expo:build, which means it works only for production because it extracted native folders and create plist files.
- if I run locally dev build it returns an error, which says sh like "missed NSFaceIDUsageDescription key".
- After extracting the native part locally (expo:prebuild) appeared iOS and Android folders and eventually I was able to run expo run:ios on a simulator with FaceID.

**Comment by Saralhossain** [comment]
I have published the fix from https://github.com/expo/expo/commit/5a23fbd5c411cbc28adb8582617d1c41873a05e3 to the latest version used by sdk-48: expo-local-authentication@~13.3.0. I updated all this still I'm facing this issue. 
"ios": {
      "supportsTablet": true,
      "infoPlist": {
        "NSFaceIDUsageDescription": "Allow $(PRODUCT_NAME) to use Face ID.",
        "NSCameraUsageDescription": "This app uses the camera to scan barcodes on event tickets."
    }
  },

can any one tell me solution?

**Comment by reverend-insanity** [comment]
> updated with "[5a23fbd](https://github.com/expo/expo/commit/5a23fbd5c411cbc28adb8582617d1c41873a05e3) to the latest version used by sdk-48: expo-local-authentication@~13.3.0." and still I'm facing this issue. "ios": { "supportsTablet": true, "infoPlist": { "NSFaceIDUsageDescription": "Allow $(PRODUCT_NAME) to use Face ID.", "NSCameraUsageDescription": "This app uses the camera to scan barcodes on event tickets." } },
> 
> solution please?

did you get a solution?

**Comment by harnie2275** [comment]
Please i am still having same issue.

**Comment by evgenyshenets** [comment]
With debug build Face ID doesn't work. You need to run prebuild and after android and iOS folders appeared you will be able to test it. Another suggestion it's test local build with Touch ID, it works as expected. Good luck

**Comment by favour626** [comment]
> With debug build Face ID doesn't work. You need to run prebuild and after android and iOS folders appeared you will be able to test it. Another suggestion it's test local build with Touch ID, it works as expected. Good luck 

Okay thanks. Was using the expo go app 😅.

**Comment by Saralhossain** [comment]
Yeah, I got the solution but it does work only for finger print authentication. @harnie2275  is right.

**Comment by harnie2275** [comment]
my ios uses faceID

**Comment by TomasSestak** [comment]
> my ios uses faceID

Did you run expo prebuild? Or using development-client? It won't work in expo go as mentioned in previous comments.

**Comment by harnie2275** [comment]
i am not using a macbook, tho, so if i run expo prebuild would it install on my expo go or what?

**Comment by TomasSestak** [comment]
> i am not using a macbook, tho, so if i run expo prebuild would it install on my expo go or what?

It creates native files (Info.plist which needs to be configured in order for Face ID to work) based on app.json which specifies the NSFaceIDUsageDescription. It has nothing to do with your OS. But I'm not sure if expo go can access that native files or you have to install dev-client. learn more here: https://docs.expo.dev/develop/development-builds/introduction/

**Comment by harnie2275** [comment]
Okay i would try it out and get back to you

**Comment by alex-wenner** [comment]
Is there a way to test this via expo go or is this planned? I see this is closed and I see mentioned that it is not supported, but that feels like a gap.

**Comment by FrancoTanzarella** [comment]
I'm on v13.4.1 and testing on my physical device. FaceID does not work for me. I did a prebuild as suggested above but this made no change. I get the same message as others.

I also tried adding the `NSFaceIdUsageDescription` to the plist file with no effect.

`{"error": null, "success": true, "warning": "FaceID is available but has not been configured. To enable FaceID, provide `NSFaceIDUsageDescription`."}`

```
"plugins": [
      "expo-router",
      "expo-localization",
      [
        "expo-location",
        {
          "locationWhenInUsePermission": "Allow $(PRODUCT_NAME) to use your location.",
          "isIosBackgroundLocationEnabled": true,
          "isAndroidBackgroundLocationEnabled": true
        }
      ],
      [
        "expo-local-authentication",
        {
          "faceIDPermissionPrompt": "Allow $(PRODUCT_NAME) to use Face ID"
        }
      ]
    ],

**Comment by Monkhai** [comment]
I am on expo SDK 49 and expo-local-authentication v13.4.1 and it is not working. Do we know if it is working only after a prebuild on a simulator or is there a fix for real devices with the expo-go application? @brentvatne

**Comment by vTrip** [comment]
Sounds like that this functionality is not available for use cases with testing in Expo Go.

We would need to build and run as native app in simulator/device.

Can we get confirmation on that? Perhaps some documentation is available to refer to this also?

Thanks

**Comment by brentvatne** [comment]
hi there! this indeed is not supported in expo go. we'll update the documentation accordingly. sorry for the confusion!

**Comment by jkincheloe33** [comment]
Are there plans to support FaceID on the expo go app? That seems silly to not since that's usually how we can test locally before pushing to a dev or prod environment.

**Comment by evgenyshenets** [comment]
Partially you can test it via Touch ID and simulators. It works in expo go and you can start with this solution. Then create dev builds and test it fully with Face ID.
